app/controllers/health_card_controller.rb

```diff
@@ -0,0 +1,71 @@
+require 'health_cards'
+require 'open-uri'
+
+class HealthCardController < ApplicationController
+  class << self
+    @vaccine_groups = nil
+
+    def vaccine_groups
+      @vaccine_groups ||= load_vaccine_groups
+    end
+
+    def load_vaccine_groups
+      f = File.open(Rails.root.join('app', 'assets', 'iisstandards_vax2vg.xml'))
+      @vaccine_groups = Nokogiri::XML(f)
+    end
+  end
+
+  @private_key = nil
+  @issuer = nil
+
+  skip_before_action :verify_authenticity_token, only: [:create]
+
+  def index; end
+
+  def create
+    # respond_to do |format|
+    #  format.fhir_json do
+    fhir_params = FHIR.from_contents(request.raw_post)
+
+    health_card = HealthCard.new
+    patient_entry = fhir_params.entry.find { |e| e.resource.is_a?(FHIR::Patient) }
+    health_card.patient = Patient.new(patient_entry.resource)
+
+    immunization_entries = fhir_params.entry.select { |e| e.resource.is_a?(FHIR::Immunization) }
+    immunization_entries.map do |immunization_entry|
```

Comment:
`map` is used when you want to return an array with mutated values, so you'd want to do something like `health_card.immunizations = immunization_entries.map {...}`. Here you're using `map` like `each`

Reply:
updated